pkg/s3select/select.go

```diff
@@ -276,91 +297,111 @@ func (reader *Input) processSelectReq(reqColNames []string, alias string, whereC
 					}
 				}
 			}
+
 			filtrCount++
 		}
 		counter++
 	}
+
+}
+
+//ReadIndividualRecord return a map[string]interfaces{} for each row in csv and each json for JSON document
+func readIndividualRecord(inputType SelectQuery) (map[string]interface{}, error) {
+	var individualRecord map[string]interface{}
+	var err error
+	switch inputType.(type) {
+	case *CSVInput:
+		individualRecord, err = inputType.Read()
+	case *JSONInput:
+		individualRecord, err = inputType.Read()
+	}
```

Comment:
should we return an error if the type doesn't match one of these two?

Reply:
There is no specific error in the case you mentioned.